PSReadLine/Position.cs

```diff
@@ -35,6 +35,41 @@ private static int GetBeginningOfLinePos(int current)
             return newCurrent;
         }
 
+        /// <summary>
+        /// Returns the position of the beginning of line
+        /// for the 0-based specified line number.
+        /// </summary>
+        private static int GetBeginningOfNthLinePos(int lineIndex)
+        {
+            System.Diagnostics.Debug.Assert(lineIndex >= 0 || lineIndex < _singleton.GetLogicalLineCount());
+
+            var nth = 0;
+            var index = 0;
+            var result = 0;
+
+            for (; index < _singleton._buffer.Length; index++)
+            {
+                if (nth == lineIndex)
+                {
+                    result = index;
+                    break;
+                }
+
+                if (_singleton._buffer[index] == '\n')
+                {
+                    nth++;
+                }

```

Comment:
There might be an issue in a corner case: when the last character in the buffer is `\n`, execution falls in this if block and let's say `nth` becomes `10` after the increment, then `index++` in the `for` loop executes and `index` becomes `_buffer.Length`. If `lineIndex` happens to be `10`, then the result will be set to `_buffer.Length` and gets returned. Would that cause a problem?

Reply:
No, it does not cause any problem.

The offset that corresponds to the length of the string is a valid cursor position if the last character of the buffer is `\n`. That is, the cursor is on a new empty line, at a position that does not correspond to a physical caracter yet.